src/lib/client.ts

```diff
@@ -32,110 +28,89 @@ export class Client {
-          async ({ hash, number }) => {
-            const blockNum = number.toNumber()
-            const signedBlock = await this.api.rpc.chain.getBlock(hash)
-            const preRuntime: SubPreDigest = this.api.registry.createType(
-              'SubPreDigest',
-              signedBlock.block.header.digest.logs.find((digestItem) => digestItem.isPreRuntime)
-                ?.asPreRuntime![1]);
-
-            if (preRuntime.solution.reward_address.toString() === rewardAddress) {
-              console.log("Farmed by me:", blockNum)
-              let blockReward = 0
-              const allRecords: Vec<any> =
-                await this.api.query.system.events.at(hash)
-              allRecords.forEach((record) => {
-                const { section, method, data } = record.event
-                if (section === "rewards" && method === "BlockReward") {
-                  const reward: u128 = this.api.registry.createType(
-                    "u128",
-                    data[1]
-                  )
-                  blockReward = Number((reward.toBigInt() * 100n) / SUNIT) / 100
-                } else if (section === "transactionFees") {
-                  // TODO
-                }
-              })
-
-              const block: FarmedBlock = {
-                id: hash.toString(),
-                time: Date.now(),
-                transactions: 0,
-                blockNum,
-                blockReward,
-                feeReward: 0,
-                rewardAddr: rewardAddress.toString(),
-                appsLink: appsLink + blockNum.toString()
-              }
-              this.data.farming.farmed = [block].concat(
-                this.data.farming.farmed
-              )
-              storeBlocks(this.data.farming.farmed)
-              this.data.farming.events.emit("farmedBlock", block)
-
-            }
-            this.data.farming.events.emit("newBlock", blockNum)
-          }
-        )
-        process.on("beforeExit", this.do.blockSubscription.stopOnReload)
-        window.addEventListener(
-          "unload",
-          this.do.blockSubscription.stopOnReload
-        )
-        this.clearTauriDestroy = await tauri.event.once(
-          "tauri://destroyed",
-          () => {
-            console.log("Destroyed event!")
-            storeBlocks(this.data.farming.farmed)
+  // TODO: implement unit test
+  // TODO: refactor using reduce
+  async getBlockRewards(hash: IU8a): Promise<number> {
+    let blockReward = 0;
+    const apiAt = await this.api.at(hash);
+    const records = (await apiAt.query.system.events()) as Vec<EventRecord>;
+    records.forEach((record) => {
+      const { section, method, data } = record.event
+      if (section === "rewards" && method === "BlockReward") {
+        const reward = this.api.registry.createType("u128", data[1]);
+        blockReward = Number((reward.toBigInt() * 100n) / SUNIT) / 100;
+      } else if (section === "transactionFees") {
```

Comment:
Yes, rewards pallet has a few different rewards all in one place, including storage and compute fees (but those are negligible).

Reply:
I was confused about that mysterious TODO, thanks for clarifying